django/core/management/commands/shell.py

```diff
@@ -111,10 +115,52 @@ def python(self, options):
         # Start the interactive interpreter.
         code.interact(local=imported_objects)
 
+    def get_and_report_namespace(self, verbosity):
+        namespace = self.get_namespace()
+
+        if verbosity == 1:
+            self.stdout.write(
+                f"{len(namespace)} objects imported automatically",
+                self.style.SUCCESS,
+            )
+        elif verbosity >= 2:
+            imports_by_module = {}
+            for obj_name, obj in namespace.items():
+                if (
+                    hasattr(obj, "__module__")
+                    and hasattr(obj, "__qualname__")
+                    and obj.__qualname__.find(".") == -1
+                ):
```

Comment:
Is there something we might be able to write instead if this condition is false?

Say someone customizes the command with:
```py
        from django.db import models
        return {
            **namespace,
            "models": models,
        }
```
gives:
--verbosity 1: prints count incremented by 1
--verbosity 2: no change

Or should we just take the stance that it's on the customizer to customize here?
***
(Apologies for the piecemeal review, I just pulled this down to test it and kept on playing with it because it's great!)

Reply:
Even without customization, you also have access to things like `auth_models`, which I didn't see from the verbose output and only discovered by reading the code.

Reply:
Originally, the --verbosity level of 2 listed only the models and functions imported, but not the modules. We decided to list everything imported in #18560, where you can see all the actual imports. What I’m unsure about is whether those changes also need to be included in this PR. If so, I will update this branch.

Reply:
I think include it here 👍